controllers/validator.go

```diff
@@ -48,11 +45,27 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
 		return validVsl, err
 	}
 
+	// ENSURE UPGRADES --------------------------------------------------------
 	// check for VSM/Volsync DataMover (OADP 1.2 or below) syntax
 	if dpa.Spec.Features != nil && dpa.Spec.Features.DataMover != nil {
 		return false, errors.New("Delete vsm from spec.configuration.velero.defaultPlugins and dataMover object from spec.features. Use Velero Built-in Data Mover instead")
 	}
 
+	// check for ResticConfig (OADP 1.4 or below) syntax
+	if dpa.Spec.Configuration.Restic != nil {
+		return false, errors.New("Delete restic object from spec.configuration, use spec.configuration.nodeAgent instead")
+	}
+	// ENSURE UPGRADES --------------------------------------------------------
+
+	// DEPRECATIONS -----------------------------------------------------------
+	if dpa.Spec.Configuration.NodeAgent != nil && dpa.Spec.Configuration.NodeAgent.UploaderType == "restic" {
```

Comment:
putting it here logs and creates an event every minute. Is this ok?

Reply:
Hmm. Wont't just one event be sufficient ? Along with log wanr and DPA warning ?

Reply:
I think so, wanted to move code to here, as it seems more appropriate place to do deprecation validation, but it has this drawback :( 

(also, only event is sufficient? or its relevant to also log the info?)

Reply:
Lets do both